doc/api/n-api.md

```diff
@@ -121,6 +121,24 @@ must use Node-API exclusively by restricting itself to using
 and by checking, for all external libraries that it uses, that the external
 library makes ABI stability guarantees similar to Node-API.
 
+### Enum values in ABI stability
+
+All enum data types defined in Node-API should be considered as a fixed size
+integer value. Bit flag enum types should be explicitly documented, and they
+work with bit operators like bit-OR (`|`) as a bit value. Unless otherwise
+documented, an enum type should be considered to be extensible.
+
+For an enum type returned from an Node-API function, or as an out parameter of
+an Node-API function, the value is an integer value and an addon should
+handle unknown values. For example, an addon should have a `default` branch
+when checking `napi_status` in switch statements as new status codes can be
+introduced in newer Node.js versions.
+
+For an enum type used in an in-parameter, extended values are guarded with
+Node-API version of which version they are introduced. The result of passing
+an unknown integer value to Node-API functions is undefined unless otherwise
+documented.
```

Comment:
Just a side note: these constraints generally look good to me but they could likely benefit from some concrete examples illustrating the correct structure.

Reply:
Added an example of `napi_key_filter`.